tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py

```diff
@@ -190,30 +198,31 @@ def _fit(self,
                 defaults.device = torch.device('cuda')
 
         logger.log(15, f'Fitting Neural Network with parameters {params}...')
-        data = self._preprocess_train(X, y, X_val, y_val, num_workers=num_workers)
+        data = self._preprocess_train(X, y, X_val, y_val)
 
-        nn_metric, objective_func_name = self.__get_objective_func_name()
+        nn_metric, objective_func_name = self.__get_objective_func_name(self.stopping_metric)
         objective_func_name_to_monitor = self.__get_objective_func_to_monitor(objective_func_name)
-        objective_optim_mode = 'min' if objective_func_name in [
-            'root_mean_squared_error', 'mean_squared_error', 'mean_absolute_error', 'r2'  # Regression objectives
-        ] else 'auto'
+        objective_optim_mode = np.less if objective_func_name in [
+            'log_loss',
+            'root_mean_squared_error', 'mean_squared_error', 'mean_absolute_error', 'median_absolute_error', 'r2',  # Regression objectives
+            'pinball_loss', # Quantile objective
+        ] else np.greater
 
         # TODO: calculate max emb concat layer size and use 1st layer as that value and 2nd in between number of classes and the value
         if params.get('layers', None) is not None:
             layers = params['layers']
         elif self.problem_type in [REGRESSION, BINARY]:
             layers = [200, 100]
+        elif self.problem_type == QUANTILE:
+            base_size = max(len(self.quantile_levels) * 4, 128)
+            layers = [base_size, base_size, base_size]
         else:
-            base_size = max(len(data.classes) * 2, 100)
+            base_size = max(data.c * 2, 100)
             layers = [base_size * 2, base_size]
 
         loss_func = None
-        if self.problem_type in [BINARY, MULTICLASS] and params.get('smoothing', 0.0) > 0.0:
-            loss_func = LabelSmoothingCrossEntropy(params['smoothing'])
-
-        ps = params['ps']
-        if type(ps) != list:
-            ps = [ps]
+        if self.problem_type == QUANTILE:
+            loss_func = HuberPinballLoss(self.quantile_levels)
```

Comment:
should alpha of the Huber loss be a hyperparameter of this model?

Reply:
Yes, it would be better to define it as a hyperparameter.
As we do not explicitly noramlize the target value, the alpha value might depend on the target value distribution.